modules/core/src/main/java/org/apache/ignite/internal/MarshallerContextImpl.java

```diff
@@ -196,6 +197,11 @@ public void onMappingDataReceived(byte platformId, Map<Integer, MappedName> mars
             int typeId = e.getKey();
             String clsName = e.getValue().className();
 
+            MappedName mappedName = platformCache.get(typeId);
+
+            if (mappedName != null && !F.isEmpty(clsName) && clsName.equals(mappedName.className()))
```

Comment:
There is a logic in mergeAndWriteMapping for checking metadata discrepancies. You should return it back if cached metadata do not match received metadata.